x/claims/keeper/integration_test.go

```diff
@@ -0,0 +1,518 @@
+		Expect(poolBalance.Amount.Uint64()).To(Equal(uint64(0)))
+
+		// ensure module account has the escrow fund
+		balanceClaims := s.app.BankKeeper.GetBalance(s.ctx, claimsAddr, claimsDenom)
+		Expect(balanceClaims.Amount.Uint64()).To(Equal(uint64(totalClaimsAmount)))
+
+		s.Commit()
+
+		proposalId = govProposal(priv0)
+	})
+
+	Context("claiming before decay duration", func() {
+		var actionV int64
+		BeforeAll(func() {
+			actionV = actionValue
+		})
+
+		It("can claim ActionDelegate", func() {
+			addr := getAddr(privs[0])
+			prebalance := s.app.BankKeeper.GetBalance(s.ctx, addr, claimsDenom)
+			delegate(privs[0], 1)
+			balance := s.app.BankKeeper.GetBalance(s.ctx, addr, claimsDenom)
+			Expect(balance.Amount.Uint64()).To(Equal(uint64(actionV + int64(prebalance.Amount.Uint64()) - 1)))
+			totalClaimed += actionV
+		})
+
+		It("can claim ActionEVM", func() {
+			addr := getAddr(privs[0])
+			prebalance := s.app.BankKeeper.GetBalance(s.ctx, addr, claimsDenom)
+			sendEthToSelf(privs[0])
+			balance := s.app.BankKeeper.GetBalance(s.ctx, addr, claimsDenom)
+			Expect(balance.Amount.Uint64()).To(Equal(uint64(actionV + int64(prebalance.Amount.Uint64()))))
+			totalClaimed += actionV
+		})
+
+		It("can claim ActionVote", func() {
+			addr := getAddr(privs[1])
+			prebalance := s.app.BankKeeper.GetBalance(s.ctx, addr, params.GetClaimsDenom())
+			govVote(privs[1], proposalId)
+			balance := s.app.BankKeeper.GetBalance(s.ctx, addr, params.GetClaimsDenom())
+			Expect(balance.Amount.Uint64()).To(Equal(uint64(actionV + int64(prebalance.Amount.Uint64()))))
+			totalClaimed += actionV
+		})
+
+		It("can claim ActionIBCTransfer", Pending, func() {
+			addr := getAddr(privs[2])
+			prebalance := s.app.BankKeeper.GetBalance(s.ctx, addr, params.GetClaimsDenom())
+			performIbcTransfer(privs[2])
+			balance := s.app.BankKeeper.GetBalance(s.ctx, addr, params.GetClaimsDenom())
+			Expect(balance.Amount.Uint64()).To(Equal(uint64(actionV + int64(prebalance.Amount.Uint64()))))
+			totalClaimed += actionV
+		})
+
+		It("did not clawback to the community pool", func() {
+			// ensure community pool doesn't have the fund
+			poolBalance := s.app.BankKeeper.GetBalance(s.ctx, distrAddr, claimsDenom)
+			Expect(poolBalance.Amount.Uint64()).To(Equal(uint64(0)))
+
+			// ensure module account has the escrow fund minus what was claimed
+			balanceClaims := s.app.BankKeeper.GetBalance(s.ctx, claimsAddr, claimsDenom)
+			Expect(balanceClaims.Amount.Uint64()).To(Equal(uint64(totalClaimsAmount - totalClaimed)))
+		})
+	})
+
+	Context("claiming at 2/3 decay duration", func() {
+		var actionV int64
+
+		BeforeAll(func() {
+			actionV = actionValue / 3
+			duration := params.DecayStartTime().Sub(s.ctx.BlockHeader().Time)
+			s.CommitAfter(duration)
+			duration = params.GetDurationOfDecay() * 2 / 3
+
+			// create another proposal to vote for
+			testTime := s.ctx.BlockHeader().Time.Add(duration)
+			s.CommitAfter(duration - time.Hour)
+			proposalId = govProposal(priv0)
```

Comment:
I saw that the default voting period for a proposal is 2 weeks. Can it be increased and how?
If not, this means multiple voting proposals need to be created during the airdrop period. Like I did in this test.

Reply:
We use the standard Cosmos sdk gov module. You can overwrite the params

```
			params := s.app.GovKeeper.GetVotingParams(s.ctx)
			params.VotingPeriod = time.Hour * 24 * 365 // 365 days
			s.app.GovKeeper.SetVotingParams(s.ctx, params)
```

You might have to change the deposit period as well.